unresolved import `crate::browser::process::get_chrome_path_from_registry

windows 10
cargo run
    Updating git repository `https://github.com/mattsse/chromiumoxide`
    Updating git repository `https://github.com/sdroege/async-tungstenite`
    Updating crates.io index
    Blocking waiting for file lock on package cache
    Blocking waiting for file lock on package cache
  Downloaded input_buffer v0.4.0
  Downloaded tungstenite v0.12.0
  Downloaded 2 crates (62.1 KB) in 2.28s
    Blocking waiting for file lock on package cache
   Compiling syn v1.0.60
   Compiling winapi v0.3.9
   Compiling getrandom v0.2.2
   Compiling input_buffer v0.4.0
   Compiling rand_core v0.6.1
   Compiling rand_chacha v0.3.0
   Compiling rand v0.8.3
   Compiling nb-connect v1.0.2
   Compiling winapi-util v0.1.5
   Compiling async-process v1.0.2
   Compiling atty v0.2.14
   Compiling termcolor v1.1.2
   Compiling ctor v0.1.19
   Compiling serde_derive v1.0.123
   Compiling futures-macro v0.3.12
   Compiling thiserror-impl v1.0.23
   Compiling async-attributes v1.1.2
   Compiling value-bag v1.0.0-alpha.6
   Compiling log v0.4.14
   Compiling futures-util v0.3.12
   Compiling polling v2.0.2
   Compiling kv-log-macro v1.0.7
   Compiling tungstenite v0.12.0
   Compiling env_logger v0.7.1
   Compiling thiserror v1.0.23
   Compiling async-io v1.3.1
   Compiling which v4.0.2
   Compiling pretty_env_logger v0.4.0
   Compiling async-global-executor v2.0.2
   Compiling async-std v1.9.0
   Compiling futures-executor v0.3.12
   Compiling futures v0.3.12
   Compiling serde v1.0.123
   Compiling serde_json v1.0.62
   Compiling async-tungstenite v0.12.0 (https://github.com/sdroege/async-tungstenite#f51f0744)
   Compiling chromiumoxide_types v0.2.0 (https://github.com/mattsse/chromiumoxide?branch=main#7d7ef5a5)
   Compiling chromiumoxide_pdl v0.2.0 (https://github.com/mattsse/chromiumoxide?branch=main#7d7ef5a5)
   Compiling chromiumoxide_cdp v0.2.0 (https://github.com/mattsse/chromiumoxide?branch=main#7d7ef5a5)
   Compiling chromiumoxide v0.2.0 (https://github.com/mattsse/chromiumoxide?branch=main#7d7ef5a5)
error[E0432]: unresolved import `crate::browser::process::get_chrome_path_from_registry`
   --> C:\Users\52752\.cargo\bin\git\checkouts\chromiumoxide-b0a6fd9b32d92502\7d7ef5a\src\browser.rs:560:13
    |
560 |         use crate::browser::process::get_chrome_path_from_registry;
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `get_chrome_path_from_registry` in `process`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0432`.
error: could not compile `chromiumoxide`

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: build failed

F:\web\xiaolu\rust\greetings\web>

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -31,6 +31,9 @@ pretty_env_logger = "0.4"
 cfg-if = "1.0"
 tokio = { version = "1.1", features = ["rt", "rt-multi-thread", "time", "fs", "macros"], optional = true }
 
+[target.'cfg(windows)'.dependencies]
+winreg = "0.8"
+
 [dev-dependencies]
 quote = "1.0"
 proc-macro2 = "1.0"

--- a/src/browser.rs
+++ b/src/browser.rs
@@ -557,9 +557,7 @@ pub fn default_executable() -> Result<std::path::PathBuf, String> {
 
     #[cfg(windows)]
     {
-        use crate::browser::process::get_chrome_path_from_registry;
-
-        if let Some(path) = get_chrome_path_from_registry() {
+        if let Some(path) = get_chrome_path_from_windows_registry() {
             if path.exists() {
                 return Ok(path);
             }
@@ -567,6 +565,15 @@ pub fn default_executable() -> Result<std::path::PathBuf, String> {
     }
 
     Err("Could not auto detect a chrome executable".to_string())
+}
+
+#[cfg(windows)]
+pub(crate) fn get_chrome_path_from_windows_registry() -> Option<std::path::PathBuf> {
+    winreg::RegKey::predef(winreg::enums::HKEY_LOCAL_MACHINE)
+        .open_subkey("SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\App Paths\\chrome.exe")
+        .and_then(|key| key.get_value::<String, _>(""))
+        .map(std::path::PathBuf::from)
+        .ok()
 }
 
 /// These are passed to the Chrome binary by default.

